### PR TITLE
feature: support backup node in upstream, the weight of backup node i…

### PR DESCRIPTION
**Pull request by membphis**
…s `0`.

if the weight of upstream node is `-1`, it means that this node is a backup node.

```json
{
    "nodes": {
        "127.0.0.1:9999": 1,
        "127.0.0.1:1980": -1  -- backup node
    },
    "type": "roundrobin",
    "desc": "new upstream"
}
```

related issue: https://github.com/apache/incubator-apisix/issues/1809

### CODE REVIEW

File: conf/config.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ apisix:
   enable_admin: true
   enable_admin_cors: true         # Admin API support CORS response headers.
   enable_debug: false
-  enable_dev_mode: false          # Sets nginx worker_processes to 1 if set to true
+  enable_dev_mode: true           # Sets nginx worker_processes to 1 if set to true
```
Comment (moonming) [inline conf/config.yaml:23]: why change this configure?

Comment (membphis) [reply conf/config.yaml:23]: will revert it later

**Comment by moonming** [comment]
@membphis ping

**Comment by membphis** [comment]
I will continue this PR later. 

If the community needs this feature, please feel free to leave a message.

**Comment by moonming** [comment]
ping @membphis

**Comment by membphis** [comment]
close this PR first. if the community need it, then we can reopen and continue this PR

**Pull request closed without merging**